File: DSA/SakshamDwivedi_590013147/Day4/D4P1.c
/*This program takes the minimum element approach, pretty much what we do during Selection Sort. Here min_element is the first element which is compared to the 
other elements and the value is changed to the compared value if the compared value is smaller [Code by Saksham Dwivedi (Comments added later)]*/
#include <stdio.h>

int arr[100];
int size;

int main()
{
    printf("Enter the size of the array (Less than 100) : ");
    scanf("%d", &size);
    printf("Enter the elements of the array : ");
    for(int i = 0; i < size; i++){
        scanf("%d", &arr[i]);
    }

    int min_element = arr[0];
    for(int i = 0; i < size; i++){
        if(min_element > arr[i]){
            min_element = arr[i];
        }
    }
    printf("Smallest Element : %d", min_element);

    return 0;
}

File: DSA/SakshamDwivedi_590013147/Day4/D4P1.py
#Solution 1
#Fixed Input Array (For quick test)
print("Solution 1 Fixed Input")
arr = [4, 2, 3, 1, 5]
arr = sorted(arr)
min_element = arr[0]
print("Smallest Element : ", min_element)

#Custom Input Array (For multiple test purposes)
print("Solution 1 Custom Input")
arr = list(int(x) for x in input("Enter an array : ").split())
arr = sorted(arr)
min_element = arr[0]
print("Smallest Element : ", min_element)



#Solution 2
#Fixed Array Input
print("Solution 2 Fixed Input")
min_element = arr[0]
for i in range(len(arr)):
    if min_element > arr[i]:
        min_element = arr[i]
print("Smallest Element : ", min_element)

#Custom Input Array
print("Solution 2 Custom Input")
arr = [4, 2, 3, 1, 5]
arr = list(int(x) for x in input("Enter an array : ").split())
min_element = arr[0]
for i in range(len(arr)):
    if min_element > arr[i]:
        min_element = arr[i]
print("Smallest Element : ", min_element)
